sdk/eventgrid/azure-messaging-eventgrid/migration-guide.md

```diff
@@ -0,0 +1,174 @@
+        "2.0"
+    ));
+}
+client.publishEvents(endpoint, getEventsList());
+```
+
+In 4.x, `publishEvents` are renamed to `sendEvents` in line with other Azure messaging SDKs such as Event Hubs and Service Bus.
+`sendEvents` doesn't need the EventGrid endpoint because the client already has it.
+```java
+List<EventGridEvent> eventsList = new ArrayList<>();
+for (int i = 0; i < 5; i++) {
+    eventsList.add(new EventGridEvent(
+        String.format("Door%d", i),
+        "Contoso.Items.ItemReceived",
+        BinaryData.fromObject(new ContosoItemReceivedEventData("Contoso Item SKU #1")),
+        "2.0"
+    ));
+}
+eventGridEventClient.sendEvents(events);
+```
+
+#### Send events to an Event Grid Domain
+To send an `EventGridEvent` to an Event Grid Domain Topic, in both versions of SDKs, you need to set the `topic` attribute for the event.
+
+In 1.x, you set the topic of an `EventGridEvent` and then send it to the topic of the Event Grid Domain.
+```java
+// Create an event 
+List<EventGridEvent> eventsList = new ArrayList<>();
+    for (int i = 0; i < 5; i++) {
+        eventsList.add(new EventGridEvent(
+            UUID.randomUUID().toString(),
+            String.format("Door%d", i),
+            new ContosoItemReceivedEventData("Contoso Item SKU #1"),
+            "Contoso.Items.ItemReceived",
+            DateTime.now(),
+            "2.0"
+        ).withTopic("myTopic"));
+    }
+```
+
+In 4.x, this remains the same except that you use `setTopic` instead of `withTopic`.
+```java
+List<EventGridEvent> eventsList = new ArrayList<>();
+for (int i = 0; i < 5; i++) {
+    eventsList.add(new EventGridEvent(
+        String.format("Door%d", i),
+        "Contoso.Items.ItemReceived",
+        BinaryData.fromObject(new ContosoItemReceivedEventData("Contoso Item SKU #1")),
+        "2.0"
+    ).setTopic("myTopic"));
+}
+```
+
+#### Deserialize events and data
+In 1.x, to consume events, you register the event types in the map and deserialize events.
+```java
+EventGridSubscriber eventGridSubscriber = new EventGridSubscriber();
+eventGridSubscriber.putCustomEventMapping("Contoso.Items.ItemReceived", ContosoItemSentEventData.class);
+EventGridEvent[] events = eventGridSubscriber.deserializeEventGridEvents(eventGridJsonString);
+EventGridEvent event = events[0];
+if (events[0].data() instanceof ContosoItemReceivedEventData){
+    ContosoItemReceivedEventData eventData=(ContosoItemReceivedEventData)events[0].data();
+    // do something
+}
+```
+
+In 4.x, you use `EventGridEvent.fromString` convenience method to deserialize `EventGridEvent` from a Json string. Then you can use the `getData()` method
+to get the event's data in a `BinaryData`, from which you can use method `toObject()` to deserialize to the data, or use `toString()` and `toBytes()`.
+```java
+List<EventGridEvent> eventGridEvents = EventGridEvent.fromString(eventGridJsonString);
+EventGridEvent event = events.get(0);
+if ("Contoso.Items.ItemReceived".equals(event.getType()) {
```

Comment:
Did track 1 support SystemEvents? If so, we should add a note here in this sample on how it would change if we are dealing with system event.

Reply:
Added the system event part. It's very similar to other events though.